code/game/objects/structures/orbital_cannon.dm

```diff
@@ -540,4 +540,5 @@
 	var/turf/target = locate(T.x + pick(-2,2), T.y + pick(-2,2), T.z)
 	sleep(15)
 	rail_gun_ammo.detonate_on(target)
+	rail_gun_ammo.ammo_count = max(0, rail_gun_ammo.ammo_count - rail_gun_ammo.ammo_used_per_firing)
```

Comment:
There should be be a check that `ammo_count > ammo_used_per_firing`

Reply:
It fires in 20/40 increments based on the ammo. and there's a check above to see if it has ammo. I think it'll be 'fine' because it can't get to fractions of 20/40.